compiler/noirc_frontend/src/node_interner.rs

```diff
@@ -1286,9 +1286,52 @@ impl NodeInterner {
                     _ => None,
                 }
             }
+            HirExpression::Constructor(expr) => {
+                let struct_type = &expr.r#type.borrow();
+
+                eprintln!("\n -> Resolve Constructor {struct_type:?}\n");
+
+                Some(struct_type.location)
+            }
+            HirExpression::MemberAccess(expr_member_access) => {
+                self.resolve_struct_member_access(expr_member_access)
+            }
+            HirExpression::Call(expr_call) => {
+                let func = expr_call.func;
+                self.resolve_location(func)
+            }
+
             _ => None,
         }
     }
+
+    fn resolve_struct_member_access(
+        &self,
+        expr_member_access: &crate::hir_def::expr::HirMemberAccess,
+    ) -> Option<Location> {
+        let expr_lhs = &expr_member_access.lhs;
+        let expr_rhs = &expr_member_access.rhs;
+
+        let found_ident = self.nodes.get(expr_lhs.into())?;
+
+        if let Node::Expression(HirExpression::Ident(ident)) = found_ident {
+            let definition_info = self.definition(ident.id);
+            if let DefinitionKind::Local(Some(local_id)) = definition_info.kind {
+                if let Some(Node::Expression(HirExpression::Constructor(constructor_expression))) =
+                    self.nodes.get(local_id.into())
+                {
+                    let struct_type = constructor_expression.r#type.borrow();
+                    let field_names = struct_type.field_names();
+                    if let Some(found) =
```

Comment:
This is four levels of nested if statements which I think hurts readability

Reply:
I've added a commit which blindly reduces the nested if statements -- lets add doc comments to this method to say what it is doing and why it is needed